x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -134,27 +140,54 @@ export class ActionExecutor {
           namespace.namespace
         );
 
-        const { actionTypeId, name, config, secrets } = actionInfo;
-        const loggerId = actionTypeId.startsWith('.') ? actionTypeId.substring(1) : actionTypeId;
-        let { logger } = this.actionExecutorContext!;
-        logger = logger.get(loggerId);
+        const { actionTypeId, name, config, secrets, rawAction, isPreconfigured } = actionInfo;
 
         if (!this.actionInfo || this.actionInfo.actionId !== actionId) {
           this.actionInfo = actionInfo;
         }
 
+        if (!actionTypeRegistry.isActionExecutable(actionId, actionTypeId, { notifyUsage: true })) {
+          actionTypeRegistry.ensureActionTypeEnabled(actionTypeId);
+        }
+        const actionType = actionTypeRegistry.get(actionTypeId);
+        const configurationUtilities = actionTypeRegistry.getUtils();
+
+        let validatedParams;
```

Comment:
Maybe this has already been discussed but would it make sense for the task types to register a `validate` function that the task manager task runner calls before calling `run()`? That way we can make sure that any validation is done at the start of task execution. Currently, this is depending on task implementers to be aware of this expectation and implement accordingly.

Reply:
Yes we discussed this and i'll do a POC for it :) 
There will be a beforeRun method on the task type that will return the raw task data (rule, action etc.)
TM will call it before calling the run method, and run validation against its result.
Then we can pass the raw result to run method, or beforeRun can keep it in the TaskRunner.

We used this approach as it looked easier but we have too much skipping logic in the subtask (IDK how should we call them).

> Currently, this is depending on task implementers to be aware of this expectation and implement accordingly.
Initially beforeRun will be optional and still needs developer's awareness.